client/ayon_core/hosts/traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -0,0 +1,722 @@
+            padding = len(basename.split("#")) - 1
+            basename = basename.replace("#" * padding, f"%0{padding}d")
+            is_sequence = True
+
+        # make absolute path to file
+        absfilepath = os.path.normpath(os.path.join(staging_dir, filepath))
+        dirname = os.path.dirname(absfilepath)
+
+        # check if dirname exists
+        if not os.path.isdir(dirname):
+            raise NotADirectoryError(
+                f"Directory '{dirname}' does not exist."
+            )
+
+        # collect all data from dirname
+        paths_for_collection = []
+        for file in os.listdir(dirname):
+            filepath = os.path.join(dirname, file)
+            paths_for_collection.append(filepath)
+
+        collections, _ = clique.assemble(paths_for_collection)
+
+        if collections:
+            collections = collections[0]
+        else:
+            if is_sequence:
+                raise ValueError(
+                    f"No collections found in directory '{dirname}'."
+                )
+
+        frame_start = None
+        frame_end = None
+        if is_sequence:
+            files = [os.path.basename(file) for file in collections]
+            frame_start = list(collections.indexes)[0]
+            frame_end = list(collections.indexes)[-1]
+        else:
+            files = basename
+
+        tags = deepcopy(repre_data["tags"])
+        # if slate in repre_data is True then remove one frame from start
+        if repre_data["slate"]:
+            tags.append("has_slate")
+
+        # get representation data
+        representation_data = {
+            "name": repre_name,
+            "ext": extension[1:],
+            "files": files,
+            "stagingDir": dirname,
+            "stagingDir_persistent": True,
+            "tags": tags,
+        }
+        if extension in VIDEO_EXTENSIONS:
+            representation_data.update({
+                "fps": repre_data["fps"],
+                "outputName": repre_name,
+            })
+
+        if explicit_output_name:
+            representation_data["outputName"] = explicit_output_name
+
+        if frame_start:
+            representation_data["frameStart"] = frame_start
+        if frame_end:
+            representation_data["frameEnd"] = frame_end
+
+        return representation_data
+
+    def _get_data_from_csv(
+        self, package_dir, filename
+    ):
+        """Generate instances from the csv file"""
+        # get current project name and code from context.data
+        project_name = self.create_context.get_current_project_name()
+
+        csv_file_path = os.path.join(
+            package_dir, filename
+        )
+
+        # make sure csv file contains columns from following list
+        required_columns = [
+            column["name"] for column in self.columns_config["columns"]
+            if column["required_column"]
+        ]
+        # get data from csv file
+        with open(csv_file_path, "r") as csv_file:
```

Comment:
You can avoid one indentation by using:
```python
with open(csv_file_path, "r") as csv_file:
    csv_content= csv_file.read()

csv_reader = csv.DictReader(StringIO(csv_content))
```
That would require import `from io import StringIO`).

Reply:
added